zilliqa/src/api/zil.rs

```diff
@@ -1074,12 +1121,59 @@ fn get_num_txns_ds_epoch(_params: Params, node: &Arc<Mutex<Node>>) -> Result<Str
     Ok(num_txns_epoch.to_string())
 }
 
+// GetTotalCoinSupply
 fn get_total_coin_supply(_params: Params, node: &Arc<Mutex<Node>>) -> Result<String> {
     let node = node.lock().unwrap();
     Ok(node.config.consensus.total_native_token_supply.to_string())
 }
 
+// GetTotalCoinSupplyAsInt
 fn get_total_coin_supply_as_int(_params: Params, node: &Arc<Mutex<Node>>) -> Result<u128> {
     let node = node.lock().unwrap();
     Ok(node.config.consensus.total_native_token_supply.0)
 }
+
+// GetMinerInfo
+fn getminerinfo(_params: Params, _node: &Arc<Mutex<Node>>) -> Result<MinerInfo> {
+    todo!("API getminerinfo is not implemented yet");
+}
+
+// GetNodeType
+fn getnodetype(_params: Params, _node: &Arc<Mutex<Node>>) -> Result<String> {
+    todo!("API getnodetype is not implemented yet");
+}
+
+// GetPrevDifficulty
+fn getprevdifficulty(_params: Params, _node: &Arc<Mutex<Node>>) -> Result<u64> {
+    todo!("API getprevdifficulty is not implemented yet");
+}
+
+// GetPrevDSDifficulty
+fn getprevdsdifficulty(_params: Params, _node: &Arc<Mutex<Node>>) -> Result<u64> {
+    todo!("API getprevdsdifficulty is not implemented yet");
+}
+
+// GetShardingStructure
+fn getshardingstructure(_params: Params, _node: &Arc<Mutex<Node>>) -> Result<()> {
+    todo!("API getshardingstructure is not implemented yet");
+}
+
+// GetSmartContractSubState
+fn getsmartcontractsubstate(_params: Params, _node: &Arc<Mutex<Node>>) -> Result<()> {
+    todo!("API getsmartcontractsubstate is not implemented yet");
+}
+
+// GetSoftConfirmedTransaction
+fn getsoftconfirmedtransaction(_params: Params, _node: &Arc<Mutex<Node>>) -> Result<GetTxResponse> {
```

Comment:
Would prefer `get_soft_confirmed_transaction` and similar for the other APIs, but happy for you to change that when you actually implement them :)

Reply:
Sure, will do when I do